### PR TITLE
Fixes #12225 - invalidate topbar cache after restart

### PR DESCRIPTION
**Pull request by lzap**
This is my third attempt. Replaces:

https://github.com/theforeman/foreman/pull/3496

### CODE REVIEW

File: app/controllers/topbar_sweeper.rb [hunk 1]
```diff
@@ -13,15 +13,15 @@ def after_destroy(record)
     record.expire_topbar_cache(self)
   end
 
-  def after_select(record)
-    record.expire_topbar_cache(self)
-  end
-
```
Comment (lzap) [inline app/controllers/topbar_sweeper.rb:19]: I can't imagine why would we invalidate cache on each select. It does not seem to be valid hook at all as well.

Comment (dLobatog) [inline app/controllers/topbar_sweeper.rb:19]: It is a valid hook, it's supposed to trigger after you select a taxonomy (https://github.com/theforeman/foreman/blob/develop/app/controllers/concerns/foreman/controller/taxonomies_controller.rb#L99)

I've just tried and if you remove it, the taxonomies selector in the top right corner doesn't changewhen you change the taxonomy, unless you remove the fragment manually. (in production)

**Comment by dLobatog** [comment]
:+1: other than the 'after_select' thing. Thanks @lzap !

**Comment by lzap** [comment]
Ok thanks, I am glad this worked out. I closed the complicated solution, this seems to be much more simple.

**Comment by dLobatog** [comment]
Merged as c3e0fed15667ef934a96c9eafddf7c094d56eab7, thanks @lzap!

**Comment by domcleal** [comment]
This PR was failing all tests and so test_develop's failing.  #3524 removes it and fixes the existing initialiser.

File: config/initializers/expire_cache.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# During restart, newly enabled or disabled plugins may modify topbar menu. For this reason,
+# cache is invalidated for all users. This will work for the default cache provider (files)
+# but might not work with other providers (memcached). On these installations, users need
```
Comment (domcleal) [inline config/initializers/expire_cache.rb:3]: "might not work"... it raises an exception: https://github.com/rails/rails/blob/v4.2.6/activesupport/lib/active_support/cache.rb#L423-L425

**Pull request closed without merging**